File: python/Mouse.py
from Komponen import Komponen
from PeripheralDevice import PeripheralDevice
class Mouse(Komponen, PeripheralDevice) :
    def __init__(self, dpi, bottonCount, refreshRate, merk, nama, interfaceType, isWireless):
        Komponen.__init__(self, merk, nama)
        PeripheralDevice.__init__(self, interfaceType, isWireless)
        
        self.__dpi = dpi
        self.__bottonCount = bottonCount
        self.__refreshRate = refreshRate
        
    def get_dpi(self):
        return self.__dpi
    def get_bottonCount(self):
        return self.__bottonCount
    def get_refreshRate(self):
        return self.__refreshRate
    
    # Setter methods
    def set_dpi(self, dpi):
        self.__dpi = dpi
    def set_bottonCount(self, bottonCount):
        self.__bottonCount = bottonCount
    def set_refreshRate(self, refreshRate):
        self.__bottonCount = refreshRate
        

File: cpp/Mouse.cpp
// Mouse.cpp (Hybrid Inheritance)
#pragma once
#include <iostream>
#include <string>
#include "Komponen.cpp"
#include "PeripheralDevice.cpp"
using namespace std;

class Mouse : public Komponen, public PeripheralDevice {
private:
    int dpi;
    int buttonCount;
    bool isGaming;
    
public:
    Mouse() {}
    
    Mouse(int dpi, int buttonCount, bool isGaming,
         string interfaceType, bool isWireless,
         string merk, string nama) : 
         Komponen(merk, nama), PeripheralDevice(interfaceType, isWireless) {
        this->dpi = dpi;
        this->buttonCount = buttonCount;
        this->isGaming = isGaming;
    }
    
    void setDpi(int dpi) {
        this->dpi = dpi;
    }
    
    void setButtonCount(int buttonCount) {
        this->buttonCount = buttonCount;
    }
    
    void setIsGaming(bool isGaming) {
        this->isGaming = isGaming;
    }
    
    int getDpi() {
        return this->dpi;
    }
    
    int getButtonCount() {
        return this->buttonCount;
    }
    
    bool getIsGaming() {
        return this->isGaming;
    }
    
    ~Mouse() {}
};
